src/main/java/com/mparticle/kits/ButtonKit.java

```diff
@@ -73,165 +66,122 @@ public ButtonKit getInstance() {
-        final String referrerValue = referrer != null ? referrer : compatReferrer;
-        if (referrerValue == null) {
-            return;
+    @Override
+    public void onResult(@Nullable Intent intent, @Nullable Throwable throwable) {
+        PackageManager pm = applicationContext.getPackageManager();
+        if (intent != null && intent.resolveActivity(pm) != null) {
+            logDebug("Handling post-install intent for %s", intent.toString());
+            AttributionResult result = new AttributionResult()
+                    .setLink(intent.getDataString())
+                    .setServiceProviderId(getConfiguration().getKitId());
+            getKitManager().onResult(result);
         }
-        ButtonLog.visibleFormat("Attribution received (Attribution token: %s)", referrerValue);
-        ButtonLog.verboseFormat(TAG, "Incoming attribution: %s", data.toString());
-        doChangeReferrer(referrerValue);
-    }
 
-    protected void doChangeReferrer(final String referrer) {
-        // We only want to track and change when the referrer actually changes.
-        if (referrer == null) {
-            return;
+        if (throwable != null) {
+            logError("Error checking post install intent", throwable);
+            AttributionError attributionError = new AttributionError()
+                    .setMessage(throwable.getMessage())
+                    .setServiceProviderId(getConfiguration().getKitId());
+            getKitManager().onError(attributionError);
         }
-        if (referrer.equals(mStorage.getReferrer())) return;
-        mStorage.setReferrer(referrer);
-        final Map<String, String> attributes = getIntegrationAttributes();
-        attributes.put(ATTRIBUTE_REFERRER, referrer);
-        ButtonLog.verboseFormat(TAG, "Updated integration attributes to: %s", attributes);
-        setIntegrationAttributes(attributes);
     }
 
+    /*
+     * Overrides for ActivityListener
+     */
+
     @Override
-    public List<ReportingMessage> onActivityCreated(final Activity activity, final Bundle savedInstanceState) {
+    public List<ReportingMessage> onActivityCreated(Activity activity, Bundle bundle) {
+        merchant.trackIncomingIntent(applicationContext, activity.getIntent());
         return null;
     }
 
     @Override
-    public List<ReportingMessage> onActivityStarted(final Activity activity) {
-        Intent intent = activity.getIntent();
-        if (intent == null) {
-            return null;
-        }
-        return handleIntent(intent.getData(), intent.getAction());
+    public List<ReportingMessage> onActivityStarted(Activity activity) {
+        merchant.trackIncomingIntent(applicationContext, activity.getIntent());
+        return null;
     }
 
     @Override
-    public List<ReportingMessage> onActivityResumed(final Activity activity) { return null; }
+    public List<ReportingMessage> onActivityResumed(Activity activity) {
+        return null;
```

Comment:
Based on our testing this would only occur if the receiving activity was already in the foreground when receiving the new intent, which seems unlikely in the case of a deep link unless it was somehow delayed or possibly triggered by a notification.

That said, I don't really see any downside in checking for a new source token in `onActivityResumed()`. Check is [added here](https://github.com/mparticle-integrations/mparticle-android-integration-button/pull/4/commits/1ade1da8279a5c957b8798ea22d35585c2149acd).

Reply:
Sounds great, thanks Chuck. I'll get this merged now, and it'll be included in our release on Wednesday